### PR TITLE
Password: Switching to text template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

###### People to notify (@mention interested parties)
@NaveenKarippai 

---

Instant Answer Page: https://duck.co/ia/view/password

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Password.pm [hunk 1]
```diff
@@ -82,12 +82,15 @@ handle remainder => sub {
     my $pw_string = join('', @pwgen);
 
     # Add password for display.
-    return $pw_string . " (random password)",
-      structured_answer => {
-        input     => [$pw_length . ' characters', $pw_strength . ' strength'],
-        operation => 'Random password',
-        result    => $pw_string
-      };
+    return $pw_string . " (random password)", structured_answer => {
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Password.pm]: `"$pw_string (random password)"`

File: t/Password.t [hunk 2]
```diff
@@ -9,162 +9,43 @@ use DDG::Test::Goodie;
...
-    ),
-    'Password High 25' => test_zci(
-        re(qr/.{25} \(random password\)/),
-        structured_answer => {
-            input     => ['25 characters', 'high strength'],
-            operation => 'Random password',
-            result    => re(qr/^.{25}$/)
-        }
-    ),
+    'pw 15 average' => build_test(re(qr/.{15}/), '15 characters, average strength'),
```
Comment (GuiltyDolphin) [inline t/Password.t]: We probably want to make these full matches `re(qr/^.{15}$/)` seeing as they represent submatches of each other.

Comment (GuiltyDolphin) [inline t/Password.t]: Hmm, actually, lets just pass the number of characters in and build the regex in the test builder.

Comment (mintsoft) [inline t/Password.t]: Much cleaner!

File: t/Password.t [hunk 3]
```diff
@@ -9,162 +9,43 @@ use DDG::Test::Goodie;
 zci answer_type => 'pw';
 zci is_cached   => 0;
 
+sub build_test {
+	my ($chars, $strength) = @_;
+	my $answer = re(qr/.{$chars}/);
```
Comment (GuiltyDolphin) [inline t/Password.t]: As before, lets wrap this in `qr/^.{$chars}$/`.

**Comment by mintsoft** [comment]
@GuiltyDolphin All done

File: t/Password.t [hunk 4]
```diff
@@ -9,162 +9,43 @@ use DDG::Test::Goodie;
 zci answer_type => 'pw';
 zci is_cached   => 0;
 
+sub build_test {
+	my ($chars, $strength) = @_;
+	my $answer = re(qr/^.{$chars}/);
```
Comment (GuiltyDolphin) [inline t/Password.t]: You missed out the `$`.

Comment (mintsoft) [inline t/Password.t]: Ah, the text answer and html are different; hence the lack of `$`

**Comment by mintsoft** [comment]
@GuiltyDolphin *now* all done

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks! :dolphin:

**Pull request merged**